serving/docs/model.md

```diff
@@ -0,0 +1,384 @@
+```
+
+To package model artifacts in a .tar.gz:
+
+```
+tar -czvf model.tar.gz /path/to/model
+```
+
+### Serving Example
+
+Let's run an example where we load a model in Python mode and run inference using the REST API.
+
+#### Step 1: Start Server
+
+First, start the server and load a model at startup. We will use the resnet18 model as an example. To get this model, first clone the djl-demo repo.
+
+```
+git clone https://github.com/deepjavalibrary/djl-demo.git
+cd djl-demo/djl-serving/python-mode
+```
+
+The resnet18 model is located at djl-serving/python-mode/resnet18. It provides a [model.py](https://github.com/deepjavalibrary/djl-demo/blob/master/djl-serving/python-mode/resnet18/model.py) that implements a `handle` function.
+
+```
+def handle(inputs: Input):
+    """
+    Default handler function
+    """
+    if not _service.initialized:
+        # stateful model
+        _service.initialize(inputs.get_properties())
+
+    if inputs.is_empty():
+        # initialization request
+        return None
+
+    return _service.inference(inputs)
+```
+
+It also provides a `requirements.txt` that loads torchvision 0.12.0:
+
+```
+torchvision==0.12.0
+```
+
+Next, run the DJL Serving and load this model at startup.
+
+Linux/macOS
+
+```
+djl-serving -m resnet::Python=file://$PWD/resnet18
+```
+
+Windows
+
+```
+path-to-your\serving.bat -m "resnet::Python=file:///%cd%\resnet18"
+```
+
+This will launch the DJL Serving Model Server, bind to port 8080, and create an endpoint named `resnet` with the model.
+
+#### Step 2: Inference
+
+To query the model using the prediction API, open another session and run the following command:
+
+Linux/macOS
+
+```
+curl -O https://resources.djl.ai/images/kitten.jpg
+curl -X POST "http://127.0.0.1:8080/predictions/resnet" -T "kitten.jpg"
+```
+
+On Windows, you can just download the image and use `Postman` to send POST request.
+
+This should return the following result:
+
+```json
+[
+  {
+    "tabby":0.4552347958087921,
+    "tiger_cat":0.3483535945415497,
+    "Egyptian_cat":0.15608155727386475,
+    "lynx":0.026761988177895546,
+    "Persian_cat":0.002232028404250741
+  }
+]
+```
+
+## Java Mode
+
+This section walks through how to serve Java based model with DJL Serving.
+
+### Define a Translator
+
+To get started, build your own Translator. The `Translator` is a Java interface defined in DJL for pre/post-processing. Your Translator should have the following signature:
+
+```
+public void prepare(TranslatorContext ctx);
+
+public NDList processInput(TranslatorContext ctx, I input) throws Exception;
+
+public String processOutput(TranslatorContext ctx, NDList list) throws Exception;
+```
+
+### Provide Model File
+
+Next, you need to include a model file. DJL Serving supports model artifacts for the following engines:
+
+- MXNet
+- PyTorch (torchscript only)
+- TensorFlow
+- ONNX
+- PadddlePaddle
+
+You can also include a `synset.txt` file in the same directory with your model file to define the labels.
```

Comment:
You can also include any required artifacts in the model directory. For example, `ImageClassificationTranslator` may need a `synset.txt` file, you can put it in the same ...

Reply:
OK